std/algorithm/comparison.d

```diff
@@ -614,40 +623,41 @@ if (isInputRange!R1 && isInputRange!R2)
         {
             if (r1.empty) return -cast(int)!r2.empty;
             if (r2.empty) return !r1.empty;
```

Comment:
For these two returns you need a bit of massaging. I'd say something like this:

```d
static if (is(typeof(r1.front.opCmp(r2.front))) R)
    alias Result = R;
else
    alias Result = int;
if (r1.empty) return -R(!r2.empty);
if (r2.empty) return R(!r1.empty);
...
```

Reply:
Fixed